src/NATS.Client/JetStream/JetStreamMsg.cs

```diff
@@ -0,0 +1,242 @@
+        /// again later.
+        /// </summary>
+        public override void Nak() => AckReply(ackNak);
+
+        /// <summary>
+        /// Prevents this message from ever being delivered regardless of
+        /// maxDeliverCount.
+        /// </summary>
+        public override void Term() => AckReply(ackTerm);
+
+        /// <summary>
+        /// Indicates that this message is being worked on and reset redelivery timer in the server.
+        /// </summary>
+        public override void InProgress() => AckReply(ackProgress);
+
+        /// <summary>
+        /// Checks if a message is from Jetstream or is a standard message.
+        /// </summary>
+        /// <returns></returns>
+        public override bool IsJetStream { get { return true; } }
+
+        /// <summary>
+        /// A JetStream message does not have a reply that is presented
+        /// to the application.
+        /// </summary>
+        public new string Reply { get { return null; } }
+    }
+
+    /// <summary>
+    /// Represents a pair of sequence numbers held by a stream and consumer.
+    /// </summary>
+    public sealed class Sequence
+    {
+        /// <summary>
+        /// Gets the Stream sequence number.
+        /// </summary>
+        public ulong StreamSequence { get; }
+
+        /// <summary>
+        /// Gets the Consumer sequence number.
+        /// </summary>
+        public ulong ConsumerSequence { get; }
+
+        internal Sequence(ulong streamSeq, ulong consSeq)
+        {
+            StreamSequence = streamSeq;
+            ConsumerSequence = consSeq;
+        }
+
+    }
+
+    /// <summary>
+    /// JetStream essage MetaData
+    /// </summary>
+    public sealed class MetaData
+    {
+        static readonly DateTime epochTime = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        /// <summary>
+        /// Number of delivered messages
+        /// </summary>
+        public ulong NumDelivered { get; }
+
+        /// <summary>
+        /// Number of pending messages
+        /// </summary>
+        public ulong NumPending { get;  }
+
+        /// <summary>
+        /// Gets the timestamp of the message.
+        /// </summary>
+        public DateTime Timestamp { get; }
+
+        /// <summary>
+        /// Gets the raw nanosecond timestamp of the message.
+        /// </summary>
+        public ulong TimestampNanos { get; }
+
+        /// <summary>
+        /// Gets the stream name.
+        /// </summary>
+        public string Stream { get;  }
+
+        /// <summary>
+        /// Gets the consumer name.
+        /// </summary>
+        public string Consumer { get;  }
+
+        /// <summary>
+        /// Gets the consumer sequence number of the message.
+        /// </summary>
+        public Sequence Sequence { get; }
+
+
+        // Caller must ensure this is a JS message
+        internal MetaData(string metaData)
+        {
+            string[] tokens = metaData?.Split('.');
+            if (tokens.Length != 8)
```

Comment:
`tokens` will be null if `metaData` is null:
```suggestion
            if (tokens?.Length != 8)
```

Reply:
Will fix...